packages/prisma/extensions/audit-log-custom-method-client-extension.ts

```diff
@@ -0,0 +1,183 @@
+import { Prisma } from "@prisma/client";
+
+import saveAuditLogWorker from "@calcom/features/audit-log/saveAuditLogWorker";
+import { BookingAuditLogOption } from "@calcom/features/audit-log/types/BookingAuditLogTypes";
+import { EventTypeAuditLogOption } from "@calcom/features/audit-log/types/EventTypeAuditLogTypes";
+import { prisma } from "@calcom/prisma";
+
+export function auditLogCustomMethodClientExtension() {
+  return Prisma.defineExtension({
+    query: {
+      user: {
+        async delete({ args, query }) {
+          const deletedUser = await query(args); //
+          const auditLogFeature = await prisma.feature.findUnique({ where: { slug: "audit-log" } });
+          if (auditLogFeature?.enabled) saveAuditLogWorker({ triggeredEvent: "user.delete", deletedUser });
+          return deletedUser; //
+        },
+      },
+      eventType: {
+        async create({ args, query }) {
+          const actorUserId = args.data?.actorUserId;
+          args.data.actorUserId = null;
+          const createdEventType = await query(args); //
+          const auditLogFeature = await prisma.feature.findUnique({ where: { slug: "audit-log" } });
+          if (auditLogFeature?.enabled && typeof actorUserId === "number") {
+            saveAuditLogWorker({
+              triggeredEvent: EventTypeAuditLogOption.EventTypeCreate,
+              actorUserId,
+              createdEventType,
+            });
+          }
+          return createdEventType; //
+        },
+        async update({ args, query }) {
+          const actorUserId = args.data?.actorUserId;
+          args.data.actorUserId = null;
+          let updatedEventType;
+          const auditLogFeature = await prisma.feature.findUnique({ where: { slug: "audit-log" } });
```

Comment:
This could be abstracted to a function: `isAuditLogEnabled`

Reply:
Done.